### PR TITLE
AoT forwarder fixes.

### PR DESCRIPTION
**Pull request by morganbr**
Fixes the ifdefs for the mscorlib and System facades to build for aot as well. Also includes a minor rd.xml tweak for BinaryFormatter. Fixes hundreds of BinaryFormatter tests. Progress on #19119

### CODE REVIEW

**Comment by morganbr** [comment]
CC @dotnet/corert-contrib , @stephentoub

File: src/shims/manual/mscorlib.forwards.cs [hunk 1]
```diff
@@ -27,6 +23,14 @@
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Collections.Generic.LongEnumEqualityComparer<>))]
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Collections.ListDictionaryInternal))]
 
+// Enum comparers that only exist in CoreCLR
```
Comment (jkotas) [inline src/shims/manual/mscorlib.forwards.cs]: They should not be needed here at all. This file should be same between CoreCLR and CoreRT.

File: src/shims/manual/mscorlib.forwards.cs [hunk 2]
```diff
@@ -5,17 +5,13 @@
 // Add any internal types that we need to forward from mscorlib. 
 
 // These types are required for Desktop to Core serialization as they are not covered by GenAPI because they are marked as internal.
-#if netcoreapp
+#if !netfx
```
Comment (jkotas) [inline src/shims/manual/mscorlib.forwards.cs]: This shim should not be building for netfx. Do we need this ifdef?

Comment (weshaggard) [inline src/shims/manual/mscorlib.forwards.cs]: No we shouldn't need the #ifdef here or in system any longer.

**Comment by weshaggard** [comment]
@morganbr did you already expose internalsvisibleto in S.P.CoreLib in corert and get that consumed in corefx? I assume so since the CI is green but just wanted to verify.

**Comment by morganbr** [comment]
@weshaggard, I made the types public instead of InternalsVisibleTo, but yes they're available in current CoreFX builds.

**Comment by morganbr** [comment]
Welp, messed up that branch. I'll open a new PR.

**Pull request closed without merging**